pkg/probe/http/http.go

```diff
@@ -28,15 +29,18 @@ import (
 )
 
 func New() HTTPProber {
-	return HTTPProber{&http.Client{}}
+	client := &http.Client{}
+	return HTTPProber{client, func(timeout time.Duration) { client.Timeout = timeout }}
 }
 
 type HTTPProber struct {
-	client HTTPGetInterface
+	client     HTTPGetInterface
+	setTimeout func(timeout time.Duration)
 }
 
 // Probe returns a ProbeRunner capable of running an http check.
-func (pr *HTTPProber) Probe(host string, port int, path string) (probe.Status, error) {
+func (pr *HTTPProber) Probe(host string, port int, path string, timeout time.Duration) (probe.Status, error) {
+	pr.setTimeout(timeout)
```

Comment:
This is no longer thread safe as the timeout can be modified by multiple callers of Probe. It's probably better to manually implement the timeout. This isn't tested and is missing some err checks but something like:
```go
func DoHTTPProbe(url string, client HTTPGetInterface, timeout time.Duration) (probe.Status, error) {
	var resp http.Response
	req, err := http.NewRequest("GET", url, nil)
	defer req.Close()
	done := make(chan struct{}, 1)
	defer close(done)
	go func() {
		resp, err = client.Do(req)
		done <- struct{}{}
	}()
	select {
	case <-done:
		break
	case <-time.After(timeout):
		return probe.Failure, fmt.Errof("Probe timed out waiting for responseresponse timed out:")
	}
	if err != nil {
		glog.V(1).Infof("HTTP probe error: %v", err)
...
```
It's not pretty but it's probably what we want in this situation since stdlib http/client doesn't have a GetTimeout() :(